editoast/src/models/timetable.rs

```diff
@@ -212,9 +214,19 @@ pub async fn get_timetable_train_schedules_with_simulations(
     let simulation_outputs = SimulationOutput::belonging_to(&train_schedules)
         .load::<SimulationOutput>(&mut conn)
         .await?;
-    let result = train_schedules
+    let mut output_map: HashMap<Option<i64>, SimulationOutput> = HashMap::new();
+
+    for output in &simulation_outputs {
+        output_map.insert(output.train_schedule_id, output.clone());
+    }
```

Comment:
This suggestion avoid to `clone()`
```suggestion
    let mut output_map: HashMap<_, _> = simulation_outputs
        .into_iter()
        .map(|o| (o.train_schedule_id.unwrap(), o))
        .collect();
```